programs/cryptid/src/instructions/propose_transaction.rs

```diff
@@ -28,19 +36,61 @@ pub struct ProposeTransaction<'info> {
             num_accounts.into(),
             InstructionSize::from_iter_to_iter(
                 instructions.iter()
-            )
-        ))
+                )
+       ))
     ]
     transaction_account: Account<'info, TransactionAccount>,
     system_program: Program<'info, System>,
 }
+
+/// Collect all accounts as a single vector so that they can be referenced by index by instructions
+// TODO: Note - I initially wanted to use some crate to iterate over a struct's fields, so I could define
+// this for all Contexts automatically, but failed. We could either leave it like this or try again.
+// Once decided, remove this comment.
+impl<'a, 'b, 'c, 'info> AllAccounts<'a, 'b, 'c, 'info>
+    for Context<'a, 'b, 'c, 'info, ProposeTransaction<'info>>
+{
+    fn all_accounts(&self) -> Vec<&AccountInfo<'info>> {
+        [
+            self.accounts.cryptid_account.as_ref(),
+            self.accounts.did.as_ref(),
+            self.accounts.did_program.as_ref(),
+            self.accounts.authority.as_ref(),
+        ]
+        .into_iter()
+        .chain(self.remaining_accounts.iter())
+        .collect()
+    }
+
+    fn get_accounts_by_indexes(&self, indexes: &[u8]) -> Result<Vec<&AccountInfo<'info>>> {
+        let accounts = self.all_accounts();
+        resolve_by_index(indexes, &accounts)
+    }
+}
+
 /// Propose a transaction to be executed by a cryptid account
 /// Note - at present, there is no constraint on who can propose a transaction.
 pub fn propose_transaction<'a, 'b, 'c, 'info>(
     ctx: Context<'a, 'b, 'c, 'info, ProposeTransaction<'info>>,
+    controller_chain: Vec<u8>,
+    did_account_bump: u8,
     instructions: Vec<AbbreviatedInstructionData>,
 ) -> Result<()> {
-    ctx.accounts.transaction_account.owner = *ctx.accounts.owner.key;
+    // convert the controller chain (an array of account indices) into an array of accounts
+    // note - cryptid does not need to check that the chain is valid, or even that they are DIDs
+    // sol_did does that
+    let controlling_did_accounts = ctx.get_accounts_by_indexes(controller_chain.as_slice())?;
+
+    // Assume at this point that anchor has verified the cryptid account and did account (but not the controller chain)
+    // We now need to verify that the signer (at the moment, only one is supported) is a valid signer for the cryptid account
+    verify_keys(
+        &ctx.accounts.did,
+        Some(did_account_bump),
+        ctx.accounts.authority.to_account_info().key,
+        controlling_did_accounts,
+    )?;
+
```

Comment:
add Account::try_from for cryptid account to ensure the cryptid account is bound to the DID

Reply:
Fixed